dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java

```diff
@@ -554,6 +568,91 @@ public final void export(@Context final HttpServletRequest httpRequest,
+     * Exports a list of categories.
+     *
+     * @param httpRequest
+     * @param httpResponse
+     * @param fileInputStream
+     * @param fileMetaData
+     * @param filter
+     * @param exportType
+     * @param contextInode
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @POST
+    @Path("/_import")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Consumes({MediaType.MULTIPART_FORM_DATA})
+    public Response importCategories(@Context final HttpServletRequest httpRequest,
+            @Context final HttpServletResponse httpResponse,
+            @FormDataParam("file") final File uploadedFile,
+            FormDataMultiPart multiPart,
+            @FormDataParam("file") FormDataContentDisposition fileDetail,
+            @FormDataParam("filter") String filter,
+            @FormDataParam("exportType") String exportType,
+            @FormDataParam("contextInode") String contextInode) {
+
+        List<Category> unableToDeleteCats = null;
+        final List<FailedResultView> failedToDelete = new ArrayList<>();
+
+        try {
+            final InitDataObject initData = webResource.init(null, httpRequest, httpResponse, true,
+                    null);
+
+            final User user = initData.getUser();
+            final PageMode pageMode = PageMode.get(httpRequest);
+
+            Logger.debug(this, () -> "Importing the list of categories. " + String.format(
+                    "Request payload is : {contextInode : %s, filter : %s, exportType : %s}", contextInode,
+                    filter, exportType));
+
+            InputStream file = new FileInputStream(uploadedFile);
+            byte [] uploadedFileBytes = file.readAllBytes();
+
+            String content = new String(uploadedFileBytes);
+            StringReader sr = new StringReader(content);
+            BufferedReader br = new BufferedReader(sr);
+
+            if(exportType.equals("replace")) {
+                Logger.debug(this, () -> "Replacing categories");
+                if(UtilMethods.isSet(contextInode)) {
+                    Category contextCat = this.categoryAPI.find(contextInode, user, false);
+                    unableToDeleteCats = this.categoryAPI.removeAllChildren(contextCat, user, false);
+                    if (!unableToDeleteCats.isEmpty()) {
+                        for (final Category category : unableToDeleteCats) {
+                            Logger.error(this, "Category with Id: " + category.getInode() + " unable to delete");
+                            failedToDelete.add(new FailedResultView(category.getInode(),
+                                    "Category with id: " + category.getInode() + " unable to delete"));
+                        }
+                    }
+                } else {
+                    Logger.debug(this, () -> "Deleting all the categories");
+                    categoryAPI.deleteAll(user, false);
```

Comment:
@jdotcms this should be also wrapped in transaction because of the `addOrUpdateCategory` below fails, the categories will be removed

Reply:
Nice catch @dsilvam 
Code has been updated to wrap things in a transaction :)